components/Slider.js

```diff
@@ -0,0 +1,246 @@
+          const stepMultiplied = step * multiplier;
+          const stepSize = (stepMultiplied / range) * 100;
+          left = valuePercentage + (stepSize * direction);
+          newValue = Number(value) + (stepMultiplied * direction);
+        }
+      }
+      if (type === 'mousemove' || type === 'click' || type === 'touchmove') {
+        const clientX = evt.touches ? evt.touches[0].clientX : evt.clientX;
+        const track = this.track.getBoundingClientRect();
+        const unrounded = ((clientX - track.left) / track.width);
+        const rounded = Math.round(((range * unrounded) / step)) * step;
+        left = (((rounded - min) / range) * 100);
+        newValue = rounded;
+      }
+    }
+
+    if (newValue <= Number(min)) {
+      left = 0;
+      newValue = min;
+    }
+    if (newValue >= Number(max)) {
+      left = 100;
+      newValue = max;
+    }
+
+    return { left, newValue };
+  }
+
+  handleMouseStart = () => {
+    this.element.ownerDocument.addEventListener('mousemove', this.updatePosition)
+    this.element.ownerDocument.addEventListener('mouseup', this.handleMouseEnd)
+  }
+
+  handleMouseEnd = () => {
+    this.element.ownerDocument.removeEventListener('mousemove', this.updatePosition)
+    this.element.ownerDocument.removeEventListener('mouseup', this.handleMouseEnd)
+  }
+
+  handleTouchStart = () => {
+    this.element.ownerDocument.addEventListener('touchmove', this.updatePosition);
+    this.element.ownerDocument.addEventListener('touchup', this.handleTouchEnd);
+    this.element.ownerDocument.addEventListener('touchend', this.handleTouchEnd);
+    this.element.ownerDocument.addEventListener('touchcancel', this.handleTouchEnd);
+  }
+
+  handleTouchEnd = () => {
+    this.element.ownerDocument.removeEventListener('touchmove', this.updatePosition);
+    this.element.ownerDocument.removeEventListener('touchup', this.handleTouchEnd);
+    this.element.ownerDocument.removeEventListener('touchend', this.handleTouchEnd);
+    this.element.ownerDocument.removeEventListener('touchcancel', this.handleTouchEnd);
+  }
+
+  renderChildren() {
+    return React.Children.map(this.props.children, (child) => {
+      return React.cloneElement(child, {
+        value: this.state.value
+      })
+    });
+  }
+
+  render() {
+    const {
+      className,
+      id,
+      onChange,
+      min,
+      minLabel,
+      max,
+      maxLabel,
+      step,
+      required,
+      disabled,
+    } = this.props;
+
+    const {
+      value,
+      left,
+    } = this.state;
+
+    const sliderClasses = classNames(
+      'bx--slider',
+      { 'bx--slider--disabled': disabled },
+      className,
+    );
+
+    const filledTrackStyle = {
+      transform: `translate(0%, -50%) scaleX(${left / 100})`,
+    }
+    const thumbStyle = {
+      left: `${left}%`,
+    }
+    return (
+      <div className="bx--slider-container">
+        <span className="bx--slider__range-label">{min}{minLabel}</span>
+        <div
+          id={id}
```

Comment:
I think the `id` needs to go on the `<input type="hidden">` in order for this to be properly associated with a `<label>`. 

For example, I=in your `SliderStory.js`:

```js
<label htmlFor="slider" className="bx--label">Slider Label</label>
<Slider id="slider" value={50} min={0} max={100} step={1} onChange={(evt) => { console.log(evt)}}>
  <TextInput id="input-for-slider" className="bx-slider-text-input"/>
</Slider>
```

We want users to be able to click on the `label` element to access the `input` and this should also make it more predictable for screen-readers to know how to navigate and input data into a form. 

Resources:
- https://stackoverflow.com/a/3051510/2058360